test/wallet-test.js

```diff
@@ -2130,13 +2131,52 @@ describe('Wallet', function() {
         await mineBlock();
 
         await forValue(node.chain, 'height', height + 1);
-        assert.equal(wdb.height, height);
+        assert.equal(wdb.height, height + 1);
 
         height += 1;
       }
 
       for (let i = 0; i < 10; i++)
         await raceForward();
     });
+
+    it('should emit details with correct confirmation', async () => {
+      const wclient = new WalletClient({port: ports.wallet});
+      await wclient.open();
+
+      const info = await wclient.createWallet('test');
+      const wallet = wclient.wallet('test', info.token);
+      await wallet.open();
+
+      const acct = await wallet.getAccount('default');
+      const waddr = acct.receiveAddress;
+
+      miner.addresses.length = 0;
+      miner.addAddress(waddr);
+
+      let txCount = 0;
+      let confirmedCount = 0;
+
+      wallet.on('tx', (details) => {
+        if (details.confirmations === 1)
+          txCount += 1;
+      });
+
+      wallet.on('confirmed', (details) => {
+        if (details.confirmations === 1)
+          confirmedCount += 1;
+      });
+
+      for (let i = 0; i < 101; i++)
+        await mineBlock();
+
+      await wallet.send({outputs: [{address: waddr, value: 1 * 1e8}]});
```

Comment:
Wouldn't this emit a `'tx'` event with 0 confirmations just before the block is mined? Because it is an outgoing TX? I wonder if its worth catching that case in the listener above with `if (details.confirmations === 0)...`

Reply:
Yeah, possibly, that may be another test case though. This one is about confirmations. We will likely remove the 'tx' event from this test case once there are 'confirmed' events for cases where the tx didn't exist in the wallet previously and the block was the first time that tx was seen (such as a coinbase in this test).

Reply:
Added to assert confirmation count is zero when it should be.